### DISCUSSION TITLE
Difficulty connecting to

### DISCUSSION THREAD

**Discussion by archiero** [apache/airflow#28420] [Q&A]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

I'm working with Apache-Airflow 2.2.2 using both MWAA in Amazon and on my local Lubuntu machine (booting from a USB stick on a Windows 10 Surface Pro). I followed the guide below to connect to Snowflake and it didn't work either place with the exact same error message of "[Errno -2] Name or service not known". I've attached the log file from my machine, the dag that fails, and requirements file for both. 

Side note, you'll notice that I have a Python operator that connects using the snowflake.connector module to log into the database. It works just fine but my superiors have said that they'd prefer the SnowflakeHook or SnowflakeOperator

https://community.snowflake.com/s/article/How-to-connect-Apache-Airflow-to-Snowflake-and-schedule-queries-jobs
[ExampleFailureSnowflake.log](https://github.com/apache/airflow/files/10248044/ExampleFailureSnowflake.log)
[requirements.txt](https://github.com/apache/airflow/files/10248047/requirements.txt)
[test_snowflake_connection_dag.txt](https://github.com/apache/airflow/files/10248060/test_snowflake_connection_dag.txt)

### What you think should happen instead

It should connect to Snowflake

### How to reproduce

Run the dag with the same requirements file

### Operating System

MWAA and Lubuntu 22.10 on USB stick with main hardrive of Windows 10 pro

### Versions of Apache Airflow Providers

2.2.2

### Deployment

MWAA

### Deployment details

MWAA (main) and virtualenv (Python 3.10) on the Lubuntu stick

### Anything else

Literally every time I attempt to use SnowflakeHook and SnowflakeOperator but never when I use snowflake.connector.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Reply by potiuk** [reply]
Yes. you clearly have a problem which you need to debug. By a quick looking at the logs, it's kinda strange that your URL contain a space (%20 url encoded) but clearly you have some typo likely somewhere. But this is your task to track the typos and problems. 

Also PLEASE in the future if you have troubleshooting questions DO NOT open "bugs" in Airflow. When you open issue you are clearly warned that you shoudl not - I guess when you opened the issue you read the hints and coments? If you want a posisble help with troubleshooting you are asked to open a discussion instead.

**Reply by Taragolis** [reply]
> it's kinda strange that your URL contain a space (%20 url encoded)

@archiero you use wrong Snowflake region `US East`, the list available regions in Snowflake: https://docs.snowflake.com/en/user-guide/intro-regions.html#commercial-regions

I briefly look at code of your DAG and I would recomend also look good section in Airflow documentation: [Best Practices](https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html)
